### PR TITLE
Added workflow for building an android APK

### PR DESCRIPTION
**Pull request by SamMousa**
This should fix the build process for android as well as automate it for CI.

### CODE REVIEW

**Comment by SamMousa** [comment]
This will contain several commits, it is hard to test the CI flow locally. (Can be squashed later)

File: .github/workflows/client.yml [hunk 1]
```diff
@@ -34,3 +33,28 @@ jobs:
...
+      - run: npm install
+        working-directory: ./client/ionic
+      - run: ionic build
+        working-directory: ./client/ionic
+      - run: ionic cap sync
+        working-directory: ./client/ionic
+      - run: ./gradlew test
+        working-directory: ./client/ionic/android
+      - run: ./gradlew build
+        working-directory: ./client/ionic/android
```
Comment (advayDev1) [inline .github/workflows/client.yml]: nit: EOF newline

File: .github/workflows/client.yml [hunk 2]
```diff
@@ -34,3 +33,28 @@ jobs:
       - run: cd client/ionic && npm install
       - run: cd client/ionic && npm audit --audit-level=moderate
       - run: cd client/ionic && CI=true npm test
+  # This builds the android APK
+  build-android-debug:
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest]
```
Comment (epicfaace) [inline .github/workflows/client.yml]: Was there a particular reason why you're running this both on Ubuntu and MacOS? I'd imagine we'd add another job for building the iOS version that would have to run on MacOS, anyway.

```suggestion
        platform: [ubuntu-latest]
```

Comment (SamMousa) [reply .github/workflows/client.yml]: Yes, we want to make sure that tests run on both environments.

File: .github/workflows/client.yml (hunk 1 above)
Comment (hspinks) [reply .github/workflows/client.yml]: simplest fix for this going forward is Editorconfig file if we want to add

**Pull request merged**